tsl/src/compression/compress_utils.c

```diff
@@ -552,20 +552,22 @@ tsl_recompress_remote_chunk(Chunk *uncompressed_chunk, FunctionCallInfo fcinfo)
 	FunctionCallInvoke(decompfn_fcinfo);
 	if (decompfn_fcinfo->isnull)
 	{
-		ereport(WARNING,
+		ereport((if_compressed ? NOTICE : ERROR),
 				(errcode(ERRCODE_INTERNAL_ERROR),
-				 errmsg("decompression failed for recompress_chunk for %u",
-						uncompressed_chunk->fd.id)));
+				 errmsg("decompression failed for recompress_chunk for %u status=%d",
+						uncompressed_chunk->fd.id,
+						uncompressed_chunk->fd.status)));
```

Comment:
I would suggest putting the status in a detailed message. I don't think the user will understand what this integer means.

Other than that, I am wondering if we shouldn't also reword this error. The current formulation sounds as if there was a failure in decompressing the function "recompress_chunk", although a chunk ID is given. I would expect the error to be about the chunk that failed, not the function. It is already obvious what function is being called. Ideally, we would also print the chunk name and not the internal ID.

Suggestion:
```suggestion
				 errmsg("decompression failed for chunk \"%s\"", get_rel_name(uncompressed_chunk->table_id),
				 errdetail("The compression status for the chunk is %d", uncompressed_chunk->fd.status)));
				 
```

Reply:
fixed